<connection>
   <sender>valueLineEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>ModernWidget</receiver>
   <slot>onLineEditChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>350</y>
    </hint>
    <hint type="destinationlabel">
     <x>200</x>
     <y>250</y>
    </hint>
   </hints>
  </connection><?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ModernWidget</class>
 <widget class="QWidget" name="ModernWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>500</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Modern Control Widget</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QWidget {
    background-color: #2b2b2b;
    color: #ffffff;
    font-family: 'Segoe UI', Arial, sans-serif;
}

QDial {
    background-color: transparent;
    border: 3px solid #404040;
    border-radius: 120px;
}

QDial::handle {
    background-color: #00aaff;
    border: 2px solid #ffffff;
    border-radius: 8px;
    width: 16px;
    height: 16px;
}

QSlider::groove:horizontal {
    border: 1px solid #404040;
    height: 8px;
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 #404040, stop:1 #202020);
    margin: 2px 0;
    border-radius: 4px;
}

QSlider::handle:horizontal {
    background: qlineargradient(x1:0, y1:0, x2:1, y2:1, stop:0 #00aaff, stop:1 #0088cc);
    border: 1px solid #ffffff;
    width: 18px;
    margin: -2px 0;
    border-radius: 9px;
}

QSlider::handle:horizontal:hover {
    background: qlineargradient(x1:0, y1:0, x2:1, y2:1, stop:0 #00ccff, stop:1 #00aadd);
}

QSlider::sub-page:horizontal {
    background: qlineargradient(x1:0, y1:0, x2:1, y2:0, stop:0 #00aaff, stop:1 #0088cc);
    border: 1px solid #404040;
    height: 8px;
    border-radius: 4px;
}

QLabel {
    color: #ffffff;
    font-weight: 500;
}

QLabel#valueLabel {
    color: #00aaff;
    font-size: 24px;
    font-weight: bold;
    border: 2px solid #404040;
    border-radius: 8px;
    padding: 8px;
    background-color: #1e1e1e;
}

QLineEdit {
    background-color: #1e1e1e;
    border: 2px solid #404040;
    border-radius: 8px;
    padding: 8px 12px;
    font-size: 14px;
    color: #ffffff;
    selection-background-color: #00aaff;
}

QLineEdit:focus {
    border-color: #00aaff;
    background-color: #252525;
}

QLineEdit:hover {
    border-color: #606060;
}

QLabel#titleLabel {
    color: #cccccc;
    font-size: 16px;
    font-weight: bold;
}

QPushButton {
    background-color: #404040;
    border: 2px solid #606060;
    border-radius: 8px;
    padding: 8px 16px;
    font-size: 14px;
    font-weight: bold;
    color: #ffffff;
}

QPushButton:hover {
    background-color: #505050;
    border-color: #00aaff;
}

QPushButton:pressed {
    background-color: #303030;
}

QPushButton#quitButton {
    background-color: #cc4444;
    border-color: #ee6666;
}

QPushButton#quitButton:hover {
    background-color: #dd5555;
}</string>
  </property>
  <layout class="QVBoxLayout" name="mainLayout">
   <property name="spacing">
    <number>20</number>
   </property>
   <property name="leftMargin">
    <number>30</number>
   </property>
   <property name="topMargin">
    <number>30</number>
   </property>
   <property name="rightMargin">
    <number>30</number>
   </property>
   <property name="bottomMargin">
    <number>30</number>
   </property>
   <item>
    <layout class="QVBoxLayout" name="inputLayout">
     <property name="spacing">
      <number>10</number>
     </property>
     <item>
      <widget class="QLabel" name="inputLabel">
       <property name="text">
        <string>Direct Input</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="valueLineEdit">
       <property name="placeholderText">
        <string>Enter value (0-100)</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="inputSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>15</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QLabel" name="titleLabel">
     <property name="text">
      <string>Control Panel</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </item>
   <item>
    <spacer name="topSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>10</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <layout class="QHBoxLayout" name="dialLayout">
     <item>
      <spacer name="leftDialSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QDial" name="dial">
       <property name="minimumSize">
        <size>
         <width>240</width>
         <height>240</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>240</width>
         <height>240</height>
        </size>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
       <property name="sliderPosition">
        <number>0</number>
       </property>
       <property name="tracking">
        <bool>true</bool>
       </property>
       <property name="wrapping">
        <bool>false</bool>
       </property>
       <property name="notchTarget">
        <double>10.000000000000000</double>
       </property>
       <property name="notchesVisible">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="rightDialSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="midSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <layout class="QVBoxLayout" name="sliderLayout">
     <property name="spacing">
      <number>15</number>
     </property>
     <item>
      <widget class="QLabel" name="sliderLabel">
       <property name="text">
        <string>Fine Control</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="horizontalSlider">
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::TicksBothSides</enum>
       </property>
       <property name="tickInterval">
        <number>10</number>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="valueSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>15</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <layout class="QVBoxLayout" name="valueLayout">
     <property name="spacing">
      <number>10</number>
     </property>
     <item>
      <widget class="QLabel" name="valueText">
       <property name="text">
        <string>Current Value</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="valueLabel">
       <property name="text">
        <string>0</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="bottomSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <layout class="QHBoxLayout" name="buttonLayout">
     <item>
      <spacer name="leftButtonSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="resetButton">
       <property name="text">
        <string>Reset</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="quitButton">
       <property name="text">
        <string>Quit</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="rightButtonSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>dial</sender>
   <signal>valueChanged(int)</signal>
   <receiver>horizontalSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>180</y>
    </hint>
    <hint type="destinationlabel">
     <x>200</x>
     <y>280</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>dial</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>280</y>
    </hint>
    <hint type="destinationlabel">
     <x>200</x>
     <y>180</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dial</sender>
   <signal>valueChanged(int)</signal>
   <receiver>valueLabel</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>180</y>
    </hint>
    <hint type="destinationlabel">
     <x>200</x>
     <y>380</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
